src/parameterizations/vertical/MOM_CVMix_KPP.F90

```diff
@@ -1501,6 +1502,60 @@ subroutine KPP_NonLocalTransport_saln(CS, G, GV, h, nonLocalTrans, surfFlux, dt,
 end subroutine KPP_NonLocalTransport_saln
 
 
+!> Apply KPP non-local transport of surface fluxes for tracers
+!> other than temperature and salinity
+subroutine KPP_NonLocalTransport_passive_tracers(applyNonLocalTrans, G, GV, h, nonLocalTrans, surfFlux, &
+                                                 dt, scalar, scale_factor)
+
+  logical,                                    intent(in)    :: applyNonLocalTrans !< if True, apply computed
+                                                                                  !! term to scalar
+  type(ocean_grid_type),                      intent(in)    :: G             !< Ocean grid
+  type(verticalGrid_type),                    intent(in)    :: GV            !< Ocean vertical grid
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)),  intent(in)    :: h             !< Layer/level thickness [H ~> m or kg m-2]
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)+1), intent(in)   :: nonLocalTrans !< Non-local transport [nondim]
+  real, dimension(SZI_(G),SZJ_(G)),           intent(in)    :: surfFlux      !< Surface flux of scalar
+                                                                        !! [conc H s-1 ~> conc m s-1 or conc kg m-2 s-1]
+  real,                                       intent(in)    :: dt            !< Time-step [s]
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)),  intent(inout) :: scalar        !< Scalar (scalar units [conc])
+  real, optional,                             intent(in)    :: scale_factor  !< Scale factor to get surfFlux
+                                                                             !! into proper units
+
+  integer :: i, j, k
+  real, dimension( SZI_(G), SZJ_(G),SZK_(GV) ) :: dtracer
+  real :: scale_factor_loc
+
+  if (present(scale_factor)) then
+    scale_factor_loc = scale_factor
+  else
+    scale_factor_loc = 1.0
+  endif
+
+  dtracer(:,:,:) = 0.0
+  !$OMP parallel do default(none) shared(dtracer, nonLocalTrans, h, G, GV, surfFlux, scale_factor_loc)
+  do k = 1, GV%ke
+    do j = G%jsc, G%jec
+      do i = G%isc, G%iec
+        dtracer(i,j,k) = ( nonLocalTrans(i,j,k) - nonLocalTrans(i,j,k+1) ) / &
+                         ( h(i,j,k) + GV%H_subroundoff ) * surfFlux(i,j) * scale_factor_loc
+      enddo
+    enddo
+  enddo
+
+  !  Update tracer due to non-local redistribution of surface flux
+  if (applyNonLocalTrans) then
```

Comment:
Is it desirable to call `KPP_NonLocalTransport_passive_tracers` when `applyNonLocalTrans = False`?
If yes, perhaps `dtracer` should only be calculated if `applyNonLocalTrans = True` or its diagnostic has been requested. If not, it would make more sense if this condition was outside of `KPP_NonLocalTransport_passive_tracers`. 
The same comment applies for `KPP_NonLocalTransport_temp` and `KPP_NonLocalTransport_saln`.

Reply:
The real issue is that I haven't plumbed in any diagnostics yet - for `KPP_NonLocalTransport_temp` and `KPP_NonLocalTransport_saln`, if `applyNonLocalTrans` is false then you still compute the terms and post the data, you just don't add the terms to the tracer quantities themselves... but when I first started working on this branch I got a little stuck trying to figure out how to include diagnostics and then I didn't come back to that before opening this PR